<HTML>
<HEAD>
    <TITLE> Molecular Diversity of Endocrine Resistance - Introduction </TITLE>
    <META http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"> 
    <META http-equiv="Content-Style-Type" content="text/css">
    <LINK rel=stylesheet type="text/css" href="Styles.css">
</HEAD>
<BODY>
    <P class = "MainTextLeft">
		Reanalysis of publicly available bioinformatics datasets may provide an important source of new knowledge.  The aim of this project is to classify endocrine-resistant tumours from publicly available datasets using known multi-gene signatures relevant to endocrine resistance. This could be used to suggest molecular mechanisms associated with resistance in individual tumours and to see how different mechanisms are represented in different datasets.
	</P>
	<P class = "MainSubTitleLeft">
		The project objectives included:
	</P>
	<P class = "MainTextLeft">
		1) Search for endocrine resistant cases in available public transcriptomic datasets<br/>
		2) Search for transcriptional signatures associated with endocrine resistance<br/>
		3) Pre-process the selected datasets for later analysis<br/>
		4) Translate signatures to the namespaces of the selected datasets<br/>
		5) Design the classification algorithm<br/>
		6) Classify the resistant cases according to the molecular mechanisms represented by the signatures<br/>
		7) Present the pipeline and results on a dedicated web-site
	</P>
	<P class = "MainTextLeft">
	The pipeline has been designed and successfully applied for classification of 9 datasets using 7 transcriptional signatures.  
	</P>
	<P class = "MainSubTitleLeft">
	The designed pipeline consists of:
	</P>
	<P class = "MainTextLeft">
		1) Procedures for a manually curated selection of relevant datasets and signatures;<br/>
		2) Procedures for semi-automatic data pre-processing, allowing cross-platform analysis;<br/>
		3) A new, fully automated, classification algorithm (Iterative Consensus PAM).<br/>
	</P>
	<P class = "MainSubTitleLeft">
		The main features of the developed pipeline include:
	</P>
	<P class = "MainTextLeft">
		1) It is based on un-supervised partitioning;<br/>
		2) It allows for "non-classifiable" samples;<br/>
		3) The procedure does not require a training set;<br/>
		4) The procedure can be used in a cross-platform context (Affymetrix & Illumina).<br/>
	</P>
	<P class = "MainTextLeft">
		The summary of selected datasets and signatures, core of the classification algorithm and the classification results are presented in the respective pages of this web site. 
	</P>
	<P class = "MainTextLeft">
		The analysis presented on this web-site was completed by Dr. Alexey Larionov for his MSc project in applied bioinformatics.  The project was completed in summer 2012 in the Edinburgh Cancer Research Centre.  The project co-supervisors were Prof. David Cameron (Edinburgh University) and Dr. Sarah Morgan (Cranfield University).  Full version if the thesis is available from the Cranfield University library using this <a href=" https://unicorn.dmz.cranfield.ac.uk/uhtbin/cgisirsi/x/0/0/57/5/3?searchdata1=581847{CKEY}&searchfield1=GENERAL^SUBJECT^GENERAL^^&user_id=WEBSERVER" target="_blank"> MSc Thesis Link</a>.
	</P>
</BODY>
</HTML>
